### DISCUSSION TITLE
Is the `UFP64` math library using a binary representation?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [FuelLabs/sway-libs#65] [Q&A] [answered]
The current documentation for [`UFP64`](https://github.com/FuelLabs/sway-libs/blob/1c2bbf33e7a7ed762f90815501f849caf47a61bf/sway_libs/src/fixed_point/ufp/ufp64/ufp64.sw) does not explain what kind of number system the library uses.

Is it binary, decimal, something in between, etc.?

I can see in the source code that the [denominator](https://github.com/FuelLabs/sway-libs/blob/1c2bbf33e7a7ed762f90815501f849caf47a61bf/sway_libs/src/fixed_point/ufp/ufp64/ufp64.sw#L28) is 2^32, but I am not quite sure what to make out of that - does it mean that the library uses a 32.32-bit number representation?

For comparison, you can see how some Solidity libraries are documented: [ABDKMath64x64](https://github.com/abdk-consulting/abdk-libraries-solidity/blob/9e69beb255e2f87d67b44047dae229ba42ee9a64/ABDKMath64x64.sol) and [PRBMath](https://github.com/paulrberg/prb-math) (particularly relevant is the [`UD60x18`](https://github.com/paulrberg/prb-math/blob/0d8c865fe4f5415a9b3d910c0d03635a22a99a50/src/UD60x18.sol) type).

**Comment by PaulRBerg** [comment]
Hey @bitzoic, could you help me with an answer here? :)

**Comment by bitzoic** [comment, marked as answer]
Yes! The `UFP64` type is a 32.32-bit number representation. 

Thank you for these references, I will create an issue to see if we can use them as a reference to improve our documentation.

If you have any other questions be sure to ask on the [Fuel Forum](https://forum.fuel.network/) so it has higher visibility!
